cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java

```diff
@@ -67,9 +82,153 @@ public class Constants {
+            true),
+        USER("user"),
+        PAGE_CURRENT("page", null, PATTERN_IS_NUMBER6,
+            ERROR_MESSAGE_INVALID_NUMBER, false),
+        PAGE_ITEMS_PER_PAGE("per_page", "results",
+            PATTERN_IS_NUMBER6, ERROR_MESSAGE_INVALID_NUMBER, false),
+        PAGE_SORT("sort");
+
+        public final static int IGNORE_PATH_PARAMETER_INDEX = 3;
+
+        public static final Set<QueryParameterKey> VALID_QUERY_PARAMETERS =
+            Arrays.stream(QueryParameterKey.values())
+                .filter(f -> f.ordinal() > IGNORE_PATH_PARAMETER_INDEX)
+                .collect(Collectors.toSet());
+
+        public static final Set<String> VALID_QUERY_PARAMETER_KEYS =
+            VALID_QUERY_PARAMETERS.stream()
+                .sorted()
+                .map(QueryParameterKey::getKey)
+                .collect(Collectors.toSet());
+
+        public static final Set<String> VALID_QUERY_PARAMETER_NVA_KEYS =
+            VALID_QUERY_PARAMETERS.stream()
+                .sorted()
+                .map(QueryParameterKey::getNvaKey)
+                .collect(Collectors.toSet());
+
+        private final String pattern;
+        private final String cristinKey;
+        private final String nvaKey;
+        private final boolean encode;
+        private final String errorMessage;
+
+        QueryParameterKey(String cristinKey) {
+            this(cristinKey, null, PATTERN_IS_NON_EMPTY, null, false);
+        }
+
+        QueryParameterKey(String cristinKey, String pattern, boolean encode) {
+            this(cristinKey, null, pattern, null, encode);
+        }
+
+        QueryParameterKey(String cristinKey, String nvaKey, String pattern) {
+            this(cristinKey, nvaKey, pattern, null, false);
+        }
 
+        QueryParameterKey(String cristinKey, String nvaKey, String pattern, String errorMessage,
+                          boolean encode) {
+            this.cristinKey = cristinKey;
+            this.nvaKey = nonNull(nvaKey) ? nvaKey : cristinKey;
+            this.pattern = pattern;
+            this.encode = encode;
+            this.errorMessage = errorMessage;
+        }
+
+        public String getNvaKey() {
+            return nvaKey;
+        }
+
+        public String getKey() {
+            return cristinKey;
+        }
+
+        public String getPattern() {
+            return pattern;
+        }
+
+        public String getErrorMessage() {
+            return errorMessage;
+        }
+
+        public boolean isEncode() {
+            return encode;
+        }
+
+        public static QueryParameterKey keyFromString(String paramName, String value) {
+            var result = Arrays.stream(QueryParameterKey.values())
+                 .filter(f -> f.ordinal() > IGNORE_PATH_PARAMETER_INDEX)
+                .filter(equalTo(paramName)).collect(Collectors.toSet());
+            return result.size() == 1
+                ? result.stream().findFirst().get()
```

Comment:
is first also a valid value?

Reply:
if only one match take it, if null or more than 1 match. test content...